shared/src/main/scala/mlscript/codegen/Codegen.scala

```diff
@@ -607,6 +607,8 @@ object JSField {
   private val identifierPattern: Regex = "^[A-Za-z$][A-Za-z0-9$]*$".r
 
   def isValidIdentifier(s: Str): Bool = identifierPattern.matches(s) && !Symbol.isKeyword(s)
+
+  def makeIdentifierValid(s: Str): Str = if (isValidIdentifier(s)) s else JSLit.makeStringLiteral(s)
```

Comment:
The name of this function is wrong. It's not about making an identifier valid. It's about emitting a valid field name specification.